tests/potential/test_potential.py

```diff
@@ -519,44 +575,96 @@ def test_force(self):
+        p.coeff['1','1']['rmin'] = rmin
+        d = p.derivative(pair=('1','1'), var=x, r=0.6)
         self.assertAlmostEqual(d, 0.6)
-        d = p.derivative(pair=('1','1'), param='m', r=[0.25,0.75])
-        np.testing.assert_allclose(d, [0.0,0.75])
+        d = p.derivative(pair=('1','1'), var=x, r=[0.25,0.75])
+        np.testing.assert_allclose(d, [0.5,0.75])
 
         #test with rmax set
-        p.coeff['1','1'].update(rmin=False, rmax=1.5)
-        d = p.derivative(pair=('1','1'), param='m', r=1.0)
+        rmax = relentless.DesignVariable(value=1.5)
+        p.coeff['1','1'].update(rmin=False, rmax=rmax)
+        d = p.derivative(pair=('1','1'), var=x, r=1.0)
         self.assertAlmostEqual(d, 1.0)
-        d = p.derivative(pair=('1','1'), param='m', r=[0.25,1.75])
-        np.testing.assert_allclose(d, [0.25,0.0])
+        d = p.derivative(pair=('1','1'), var=x, r=[0.25,1.75])
+        np.testing.assert_allclose(d, [0.25,1.5])
 
         #test with rmin and rmax set
-        p.coeff['1','1']['rmin'] = 0.5
-        d = p.derivative(pair=('1','1'), param='m', r=0.75)
+        p.coeff['1','1']['rmin'] = rmin
+        d = p.derivative(pair=('1','1'), var=x, r=0.75)
         self.assertAlmostEqual(d, 0.75)
-        d = p.derivative(pair=('1','1'), param='m', r=[0.25,0.5,1.5,1.75])
-        np.testing.assert_allclose(d, [0.0,0.5,1.5,0.0])
+        d = p.derivative(pair=('1','1'), var=x, r=[0.25,0.5,1.5,1.75])
+        np.testing.assert_allclose(d, [0.5,0.5,1.5,1.5])
 
-        #test with shift set
+        #test w.r.t. rmin and rmax
+        d = p.derivative(pair=('1','1'), var=rmin, r=[0.25,1.0,2.0])
+        np.testing.assert_allclose(d, [2.0,0.0,0.0])
+        d = p.derivative(pair=('1','1'), var=rmax, r=[0.25,1.0,2.0])
+        np.testing.assert_allclose(d, [0.0,0.0,2.0])
+
+        #test parameter derivative with shift set
         p.coeff['1','1'].update(shift=True)
-        d = p.derivative(pair=('1','1'), param='m', r=0.5)
+        d = p.derivative(pair=('1','1'), var=x, r=0.5)
+        self.assertAlmostEqual(d, -1.0)
+        d = p.derivative(pair=('1','1'), var=x, r=[0.25,1.0,1.5,1.75])
+        np.testing.assert_allclose(d, [-1.0,-0.5,0.0,0.0])
+
+        #test w.r.t. rmin and rmax, shift set
+        d = p.derivative(pair=('1','1'), var=rmin, r=[0.25,1.0,2.0])
+        np.testing.assert_allclose(d, [2.0,0.0,0.0])
+        d = p.derivative(pair=('1','1'), var=rmax, r=[0.25,1.0,2.0])
+        np.testing.assert_allclose(d, [-2.0,-2.0,0.0])
+
+    def test_derivative_types(self):
+        """Test derivative method with different param types."""
+        q = LinPot(types=('1',), params=('m',))
+        x = relentless.DesignVariable(value=4.0)
+        y = relentless.DesignVariable(value=64.0)
+        z = relentless.GeometricMean(x, y)
+        q.coeff['1','1']['m'] = z
+
+        #test with respect to dependent variable parameter
+        d = q.derivative(pair=('1','1'), var=z, r=2.0)
+        #self.assertAlmostEqual(d, 2.0)
```

Comment:
Is this test supposed to be commented out?
```suggestion
        self.assertAlmostEqual(d, 2.0)
```